### ISSUE TITLE
ionide does not load projects

### ISSUE THREAD

**Issue by atheck** [ionide/ionide-vscode-fsharp#1479]
### Describe the bug**

I'm using ionide in vs code. After opening vs code ionide does not load any projects. All projects are "not loaded yet". When I open a fs file, in the status bar it says "Not in a F# project" and the error is: "command 'fsharp.AddFileToProject' not found".

Installed dotnet SDKs:
* 2.1.812 [/usr/share/dotnet/sdk]
* 3.1.405 [/usr/share/dotnet/sdk]
* 5.0.102 [/usr/share/dotnet/sdk]

Installed dotnet runtimes:
* Microsoft.AspNetCore.All 2.1.24 [/usr/share/dotnet/shared/Microsoft.AspNetCore.All]
* Microsoft.AspNetCore.App 2.1.24 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
* Microsoft.AspNetCore.App 3.1.11 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
* Microsoft.AspNetCore.App 5.0.2 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
* Microsoft.NETCore.App 2.1.24 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
* Microsoft.NETCore.App 3.1.11 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
* Microsoft.NETCore.App 5.0.2 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

Also I have installed latest mono.

Am I missing something?

### Expected behaviour
All projects should load.

## Machine info
 - OS: Ubuntu 20.04
 - .NET SDK version: see above
 - Ionide version: 5.1.0

**Comment by atheck** [comment]
Hi, it is working again. I don't know why. Maybe some update.

Issue can be closed.

**Comment by rytido** [comment]
I am having this same issue on Windows with the same SDKs installed. @atheck no idea why it started working again?

**Comment by atheck** [comment]
@rytido no sorry, I don't know why it's working again. I did nothing but ignoring it. Maybe we should reopen this issue?

**Comment by rytido** [comment]
I think so. I uninstalled all the SDKs and Ionide and reinstalled, and I still get the same issue. Thank you.

**Comment by SluttyPutty** [comment]
I unsubbed this a while ago. why am i still receivinf these.

On Jan 22, 2021, at 11:06, rytido <notifications@github.com> wrote:

﻿
I think so. I uninstalled all the SDKs and Ionide and reinstalled, and I still get the same issue. Thank you.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, <#view it on GitHub>

, or unsubscribe.

**Comment by rytido** [comment]
If I downgrade to Ionide 4.17.0 it works

**Comment by nyash** [comment]
I had an identical problem on Windows 10.

VSCode would show a pop up: Not in a F# project"  -> "command 'fsharp.AddFileToProject' not found

`dotnet --list-sdks` listed a bunch of SDKs (a lot of them old/previews), so I went to Start->Control Panel -> Programs and removed every Microsoft .NET SDK I could except versions 5.0.100 x64 (From Visual Studio) and 5.0.102 x64:

Surpringly  `dotnet --list-sdks` still listed some old SDKs but they were not present in Control Panel to remove.
So I went to: 

`C:\Program Files (x86)\dotnet\sdk` and removed folders manually from there.

Now `dotnet --list-sdks` just lists:
5.0.100 [C:\Program Files\dotnet\sdk]
5.0.102 [C:\Program Files\dotnet\sdk]

And Ionide loaded without any problems after opening VSCode.

**Comment by chuckwolber** [comment]
This did not help in macOS.  I still reliably get command 'fsharp.AddFileToProject' not found.

<img width="486" alt="Screen Shot 2021-02-07 at 12 04 30 AM" src="https://user-images.githubusercontent.com/8021037/107140629-2ea93d80-68d8-11eb-84f7-6ad1964c02c1.png">

**Comment by not-rusty** [comment]
Hello there,

I had the same issue. I had multiple js related folders in my workspace, so I just opened the f# project with vscode's `Open folder...`, and Ionide correctly loads.

Maybe the issue is working with a workspace with multiple root folders?

**Comment by joshuapassos** [comment]
Here i use remote workspace in vscode. After update to > 5.2.0 i have a similar problem. ionide not start in vscode

**Comment by baronfel** [comment]
@joshuapassos I don't know that we support 'remote workspaces' in vscode. Can you provide any information about how these remote workspaces work?

**Comment by joshuapassos** [comment]
> @joshuapassos I don't know that we support 'remote workspaces' in vscode. Can you provide any information about how these remote workspaces work?

I use a remote machine running vscode server and connect from my pc using ssh. 

This link has more descriptions
https://code.visualstudio.com/docs/remote/ssh

Ionide <= 5.2 works very well

**Comment by Richie-Jang** [comment]
Hello there,

I am having a same issue on Win10.
By the way, If I clicked "Project loading failed", I could see following messages.

Status:failed to load
Error:  (It is Korean Originally, I translate to English ..) 
Starting build. "SyncServer.fsproj"  Project (ResolvePackageDependenciesDesignTime;_GenerateCompileDependencyCache;CoreCompile 대상): 도구 버전 "Current"(을)를 사용하여 빌드하고 있습니다 {Tool Version 'Current' is using for build }. 조건이 잘못되어 { Condition is wrong} "_CheckForUnsupportedTargetFramework" 대상을 건너뜁니다. { Skip the target }   ('$(_UnsupportedTargetFrameworkError)' == 'true')이(가) ('' == 'true')(으)로 확인되었습니다 { is confirmed}. "d:\StudyProgramming\FSharp\Sockets\SyncServer\SyncServer.fsproj" 프로젝트의 "C:\Program ...

Is this helpful to solve this issue?

Thank you.

**Comment by cartermp** [comment]
@not-rusty multiple workspace roots seems like a likely cause for issues. In theory it should be able to support that, but things could start to get rather complicated (e.g., imagine some roots have other F# projects/solutions in them, which does F# Solution Explorer populate with?)

@Richie-Jang Is there more in the message that you can share? What is the `TargetFramework` of that project?

**Comment by Krzysztof-Cieslak** [comment]
@Richie-Jang

Make sure that you have installed .NET SDK version corresponding to the one in `global.json` file, and then try removing `bin` and `obj` folders, and running `dotnet restore` again

**Comment by Richie-Jang** [comment]
@Krzysztof-Cieslak  and @cartermp 
I found the reason why VSCode had failed when Project loading.
In my System, I have three Dotnet SDK (using dotnet --list-skds)
===========================================
5.0.103 [C:\Program Files\dotnet\sdk]
5.0.200-preview.21079.7 [C:\Program Files\dotnet\sdk]
6.0.100-preview.1.21103.13 [C:\Program Files\dotnet\sdk]

In Failed log file, I can see IonIde is using SDK 5.0.200-preview.21079.7
Go to directory >> C:\Program Files\dotnet\sdk\5.0.200-preview.21079.7\FSharp\
There is fsc.exe file , only existed fsc.dll

By the way, the previous version of SDK, 5.0.103, I can find fsc.exe in FSharp folder.
Also IonIde is using Exe file too.
So I copy **fsc.exe** file to SDK 5.0.200 FSharp folder and start IonIDE again, Project loading fine. Back to normal.

I hope this information is helpful to solve it.
Thanks.

**Comment by Richie-Jang** [comment]
@Krzysztof-Cieslak and @cartermp

I found the reason why VSCode had failed when Project loading.
In my System, I have three Dotnet SDK (using dotnet --list-skds)
5.0.103 [C:\Program Files\dotnet\sdk]
5.0.200-preview.21079.7 [C:\Program Files\dotnet\sdk]
6.0.100-preview.1.21103.13 [C:\Program Files\dotnet\sdk]

In Failed log file, I can see IonIde is using SDK 5.0.200-preview.21079.7
Go to directory >> C:\Program Files\dotnet\sdk\5.0.200-preview.21079.7\FSharp
There is fsc.exe file , only existed fsc.dll

By the way, the previous version of SDK, 5.0.103, I can find fsc.exe in FSharp folder.
Also IonIde is using Exe file too.
So I copy fsc.exe file to SDK 5.0.200 FSharp folder and start IonIDE again, Project loading fine. Back to normal.

I hope this information is helpful to solve it.
Thanks.

**Comment by rytido** [comment]
I also only have the issue if I have multiple folders in my workspace.

**Comment by rytido** [comment]
Not working again, even with a single folder. This is driving me nuts.

**Comment by rytido** [comment]
Ok, as @Richie-Jang indicated, 5.0.200 appears to be missing fsc.exe, which ionide expects. If I use 5.0.103 _with a single-folder workspace_, ionide is happy. However, still getting _command 'fsharp.AddFileToProject' not found_ in multi-folder workspaces.

**Comment by baronfel** [comment]
We're working the 5.0.200 issue in another thread, and we've already discussed that multiple folders are not a thing in ionide yet.

**Comment by rytido** [comment]
@baronfel, sorry, I saw some speculation but not confirmation that multiple workspaces were not supported. It worked in ionide 4 with dotnet 3.1, so is this new in 5?

**Comment by baronfel** [comment]
We've never explicitly coded in support for multiple workspaces in Ionide, so any support you did have was entirely incidental. We probably should support them, but as @cartermp mentioned above there's some thought that needs to go into the implementation, because that model doesn't really amtch with .net tooling in general.

**Comment by baronfel** [comment]
@rytido as a workaround, you can explicitly set the path of the project/solution you want to load for ionide as a workspace setting (for example `"FSharp.workspacePath": "./my_app.sln",`) and that will work correct in a multi-workspace-folder situation. I just tested this with my own work project by adding a code-workspace file and it worked great.

**Comment by rytido** [comment]
Thank you! 5.4.0 seems to have partially addressed this by instead prompting you to select a project file if it's not obvious from the given context. However, this feature has yet to work for me: nothing happens when I select the project.

**Comment by bitstormGER** [comment]
Hi, 
I just finished setting up a new Arch linux install with vscode 1.56.2 and dotnet sdk 5.0.203.
I am new to f# and ionide so I did setup a new console project and opened the folder with vscode.

I installed ionide fsharp 5.5.5 and have the dotnet extension pack 1.0.3 installed. 

As far as I can say ionide doesn't open if I open a f# file and if I try to run and debug I always get the message that I don't have a f# extension installed which will lead me to the dotnet extension pack which I have already installed.

I hope that I didn't simply make a mistake on my side and that this information is useful.
